libpromises/crypto.c

```diff
@@ -112,51 +127,39 @@ static void RandomSeed(void)
 
     /* 2. Seed the strong OpenSSL PRNG. */
 
-    /* randseed file is written by cf-key. */
+    /* randseed file is written on deinitialization of crypto system */
     char randfile[CF_BUFSIZE];
     snprintf(randfile, CF_BUFSIZE, "%s%crandseed",
              CFWORKDIR, FILE_SEPARATOR);
     Log(LOG_LEVEL_VERBOSE, "Looking for a source of entropy in '%s'",
         randfile);
 
-    if (!RAND_load_file(randfile, -1))
-    {
-        Log(LOG_LEVEL_VERBOSE,
-            "Could not read sufficient randomness from '%s'", randfile);
-    }
-
 #ifndef __MINGW32__                                     /* windows may hang */
     RAND_poll();
 #else
     RAND_screen();
 #endif
 
-    /* We should have had enough entropy by now. Else we print a message and
-     * use non-crypto-safe random data. */
     if (RAND_status() != 1)
     {
-        /* TODO raise to LOG_LEVEL_WARNING? */
-        Log(LOG_LEVEL_INFO,
-            "PRNG hasn't been seeded enough, using some pseudo-random bytes for seed!");
-        Log(LOG_LEVEL_INFO,
-            "A workaround is to copy 1KB of random bytes to '%s'",
-            randfile);
-
-        unsigned char rand_buf[128];
-        for (size_t i = 0; i < sizeof(rand_buf); i++)
+        if (RAND_load_file(randfile, -1) != 1024)
         {
-            rand_buf[i] = rand() % 256;
+            Log(LOG_LEVEL_CRIT,
+                "Could not read randomness from '%s'", randfile);
```

Comment:
I don't think printing the workaround is a good idea. I don't think the users should be trusted to come up with seed-grade entropy.